python/cugraph/cugraph/tests/utils/test_dataset.py

```diff
@@ -328,6 +328,16 @@ def test_is_multigraph(dataset):
     assert G.is_multigraph() == dataset.metadata["is_multigraph"]
 
 
+@pytest.mark.parametrize("dataset", ALL_DATASETS)
+def test_object_getters(dataset):
+    assert dataset.is_directed() == dataset.metadata["is_directed"]
+    assert dataset.is_multigraph() == dataset.metadata["is_multigraph"]
+    assert dataset.is_symmetric() == dataset.metadata["is_symmetric"]
+    assert dataset.number_of_nodes() == dataset.metadata["number_of_nodes"]
+    assert dataset.number_of_vertices() == dataset.metadata["number_of_nodes"]
+    assert dataset.number_of_edges() == dataset.metadata["number_of_edges"]
```

Comment:
This really isn't testing much more than if the APIs are returning the metadata dict, and could start failing if we ever intentionally change the implementation to return something that's still valid but doesn't match the metadata dict exactly.

A better test would be if you verified `is_directed` and others are actually True or False based on if the datasets are directed or not, etc.